code/game/gamemodes/cult/blood_magic.dm

```diff
@@ -0,0 +1,662 @@
+		if(B.button.locked)
+			var/order = pix_X + spells.Find(B) * 31
+			B.button.screen_loc = "[screen_loc_X[1]]:[order],[screen_loc_Y[1]]:[screen_loc_Y[2]]"
+			B.button.moved = B.button.screen_loc
+
+/datum/action/innate/cult/blood_magic/Activate()
+	var/rune = FALSE
+	var/limit = RUNELESS_MAX_BLOODCHARGE
+	for(var/obj/effect/rune/empower/R in range(1, owner))
+		rune = TRUE
+		break
+	if(rune)
+		limit = MAX_BLOODCHARGE
+	if(spells.len >= limit)
+		if(rune)
+			to_chat(owner, "<span class='cultitalic'>You cannot store more than [MAX_BLOODCHARGE] spells. <b>Pick a spell to remove.</b></span>")
+		else
+			to_chat(owner, "<span class='cultitalic'><b><u>You cannot store more than [RUNELESS_MAX_BLOODCHARGE] spells without an empowering rune! Pick a spell to remove.</b></u></span>")
+		var/nullify_spell = input(owner, "Choose a spell to remove.", "Current Spells") as null|anything in spells
+		if(nullify_spell)
+			qdel(nullify_spell)
+		return
+	var/entered_spell_name
+	var/datum/action/innate/cult/blood_spell/BS
+	var/list/possible_spells = list()
+	for(var/I in subtypesof(/datum/action/innate/cult/blood_spell))
+		var/datum/action/innate/cult/blood_spell/J = I
+		var/cult_name = initial(J.name)
+		possible_spells[cult_name] = J
+	possible_spells += "(REMOVE SPELL)"
+	entered_spell_name = input(owner, "Pick a blood spell to prepare...", "Spell Choices") as null|anything in possible_spells
+	if(entered_spell_name == "(REMOVE SPELL)")
+		var/nullify_spell = input(owner, "Choose a spell to remove.", "Current Spells") as null|anything in spells
+		if(nullify_spell)
+			qdel(nullify_spell)
+		return
+	BS = possible_spells[entered_spell_name]
+	if(QDELETED(src) || owner.incapacitated() || !BS || (rune && !(locate(/obj/effect/rune/empower) in range(1, owner))) || (spells.len >= limit))
+		return
+	to_chat(owner,"<span class='warning'>You begin to carve unnatural symbols into your flesh!</span>")
+	SEND_SOUND(owner, sound('sound/weapons/slice.ogg',0,1,10))
+	if(!channeling)
+		channeling = TRUE
+	else
+		to_chat(owner, "<span class='cultitalic'>You are already invoking blood magic!</span>")
+		return
+	if(do_after(owner, 100 - rune * 60, target = owner))
+		if(ishuman(owner))
+			var/mob/living/carbon/human/H = owner
+			H.bleed(20 - rune * 12)
+		var/datum/action/innate/cult/blood_spell/new_spell = new BS(owner)
+		new_spell.Grant(owner, src)
+		spells += new_spell
+		Positioning()
+		to_chat(owner, "<span class='warning'>Your wounds glow with power, you have prepared a [new_spell.name] invocation!</span>")
+	channeling = FALSE
+
+/datum/action/innate/cult/blood_spell //The next generation of talismans, handles storage/creation of blood magic
+	name = "Blood Magic"
+	button_icon_state = "telerune"
+	background_icon_state = "bg_demon"
+	desc = "Fear the Old Blood."
+	var/charges = 1
+	var/magic_path = null
+	var/obj/item/melee/blood_magic/hand_magic
+	var/datum/action/innate/cult/blood_magic/all_magic
+	var/base_desc //To allow for updating tooltips
```

Comment:
can use initial(desc) instead of saving it separately

Reply:
base_desc is modified and is not equal to initial(desc)